packages/core/src/node-utils/files.ts

```diff
@@ -165,6 +166,17 @@ function flushDirectories(directory: string, recursive = true): void {
   } catch (e) {}
 }
 
+async function readFileAndCreateHash(filePath: string): Promise<string> {
+  return fs.promises
+    .readFile(filePath, "utf-8")
+    .then((file) => createHash("md5").update(file).digest("hex"));
+}
+
+function readFileAndCreateHashSync(filePath: string): string {
```

Comment:
why do we have a sync and an async method for computing the hash?

Reply:
I Wrote the async one first, but then found another place where it would be needed in a blocking function here https://github.com/prismicio/slice-machine/pull/573/files#diff-7dedffd7da0ef7017990f231207cbbe1adb822b8f00dff1d78398daeaeaaf954R55

`export function resolvePathsToScreenshot({` so it was a choice of either refactoring that file or adding a sync version.